purchases/src/androidTestIntegrationTest/kotlin/com/revenuecat/purchases/PurchasesIntegrationTest.kt

```diff
@@ -0,0 +1,190 @@
+import io.mockk.mockk
+import io.mockk.verify
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.fail
+import org.junit.After
+import org.junit.Before
+import org.junit.BeforeClass
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.net.URL
+import java.util.Date
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+@RunWith(AndroidJUnit4::class)
+class PurchasesIntegrationTest {
+
+    companion object {
+        @BeforeClass @JvmStatic
+        fun setupClass() {
+            if (!canRunIntegrationTests()) {
+                error("You need to set required constants in Constants.kt")
+            }
+        }
+
+        private const val testTimeoutInSeconds = 5L
+
+        private fun canRunIntegrationTests() = Constants.apiKey != "REVENUECAT_API_KEY" &&
+            Constants.googlePurchaseToken != "GOOGLE_PURCHASE_TOKEN" &&
+            Constants.productIdToPurchase != "PRODUCT_ID_TO_PURCHASE"
+    }
+
+    private val currentTimestamp = Date().time
+    private val testUserId = "android-integration-test-$currentTimestamp"
+    private val proxyUrl = Constants.proxyUrl.takeIf { it != "NO_PROXY_URL" }
+
+    private lateinit var mockBillingAbstract: BillingAbstract
+
+    private var latestPurchasesUpdatedListener: BillingAbstract.PurchasesUpdatedListener? = null
+    private var latestStateListener: BillingAbstract.StateListener? = null
+
+    @get:Rule
+    var activityScenarioRule = activityScenarioRule<MainActivity>()
+
+    @Before
+    fun setup() {
+        latestPurchasesUpdatedListener = null
+        latestStateListener = null
+
+        onActivityReady {
+            clearAllSharedPreferences(it)
+
+            mockBillingAbstract = mockk<BillingAbstract>(relaxed = true).apply {
+                every { purchasesUpdatedListener = any() } answers { latestPurchasesUpdatedListener = firstArg() }
+                every { stateListener = any() } answers { latestStateListener = firstArg() }
+            }
+
+            proxyUrl?.let { urlString ->
+                Purchases.proxyURL = URL(urlString)
+            }
+
+            Purchases.configure(
+                PurchasesConfiguration.Builder(it, Constants.apiKey)
+                    .appUserID(testUserId)
+                    .build(),
+                mockBillingAbstract
+            )
+        }
+    }
+
+    @After
+    fun tearDown() {
+        Purchases.sharedInstance.close()
+    }
+
+    // region tests
+
+    @Test
+    fun sdkCanBeConfigured() {
+        onActivityReady {
+            assertThat(Purchases.sharedInstance.appUserID).isNotNull
+        }
+    }
+
+    @Test
+    fun customerInfoCanBeFetched() {
+        val lock = CountDownLatch(1)
+
+        onActivityReady {
+            Purchases.sharedInstance.getPurchaserInfoWith(
+                onError = { fail("should be success. Error: ${it.message}") },
+                onSuccess = { lock.countDown() }
```

Comment:
wait, how is this working? This should 5xx, right?

Reply:
That request is not being redirected to the load shedder, that's why it works. In any case, that test isn't needed for the load shedder so I removed it.